File: go/isAnagram_test.go
// https://leetcode.com/problems/valid-anagram

package leetcode

import "testing"

func Test_isAnagram(t *testing.T) {
	type args struct {
		s string
		t string
	}
	tests := []struct {
		name string
		args args
		want bool
	}{
		{
			name: "Example 1",
			args: args{"anagram", "nagaram"},
			want: true,
		},
		{
			name: "Example 2",
			args: args{"rat", "car"},
			want: false,
		},
		{
			name: "Not same length",
			args: args{"horse", "sore"},
			want: false,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := isAnagram(tt.args.s, tt.args.t); got != tt.want {
				t.Errorf("isAnagram() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: c/src/isAnagram_test.c
#include <assert.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

#include "isAnagram.c"

typedef struct test {
    char* s;
    char* t;
    bool want;
} test;

int main(int argc, char* argv[])
{
    test tests[] = {
        { "anagram", "nagaram", true },
        { "rat", "car", false },
    };
    size_t len_tests = sizeof(tests) / sizeof(test);

    for (size_t i = 0; i < len_tests; i++) {
        test* t = &tests[i];
        bool got = isAnagram(t->s, t->t);
        assert(got == t->want);
    }

    exit(EXIT_SUCCESS);
}
